core/tests/protractor_desktop/additionalPlayerFeatures.js

```diff
@@ -165,8 +165,11 @@ describe('Full exploration editor', function() {
     await general.moveToPlayer();
     await explorationPlayerPage.submitAnswer('Continue');
     var backButton = element(by.css('.protractor-test-back-button'));
+    var nextCardButton = element(by.css('.protractor-test-next-card-button'));
     expect(await backButton.isPresent()).toEqual(true);
     await explorationPlayerPage.submitAnswer('LogicProof');
+    await waitFor.visibilityOf(
+      nextCardButton, 'Next Card button taking too long to show up.');
```

Comment:
The `explorationPlayerPage.clickThroughToNextCard` function includes a waitFor and uses action.click:

https://github.com/oppia/oppia/blob/5d3a83620bb3fdf4b9e6dab03658bdee12510744/core/tests/protractor_utils/ExplorationPlayerPage.js#L120-L123

However, it refers to a different class than the `protractor-test-next-card-button` you use. Could you investigate what the difference is between these two buttons and why we need to wait for this other one?

Reply:
They're actually the same button, the class I used just points to a `<span>` inside the button used by `explorationPlayerPage.clickThroughToNextCard()`. I can change the class I use in `additionalPlayerFeatures` if you'd like me to.

I want to wait for the button to show up since the visibility of the back button is tied to the visibility of the `nextCardButton` via an angular variable (i.e. when the `nextCardButton` appears, the back button disappears). By first checking that the `nextCardButton` is visible and then checking that the back button is invisible, I think we're reducing the chance of the test flaking.